mlflow/spark.py

```diff
@@ -166,14 +167,33 @@ def log_model(
                       The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
                       from datasets with valid model input (e.g. the training dataset with target
                       column omitted) and valid model output (e.g. model predictions generated on
-                      the training dataset), for example:
+                      the training dataset). The output schema defined in a signature will filter
+                      the model prediction return fields to return only those defined in the
+                      signature. For example:
 
-                      .. code-block:: python
+    .. code-block:: python
+
+        from mlflow.models.signature import infer_signature
+        train = df.drop("target_label")
+        predictions = ... # compute model predictions
+        signature = infer_signature(train, predictions)
+
+        # define the output signature for returned predictions from a `predict()` transformation
+        mlflow.spark.log_model(model, "model", signature=signature, registered_model_name="mymodel")
+
+    .. code-block:: bash
+
+        $ mlflow models serve -m models:/mymodel/1 -p 5000
+
+        # model predictions columns will correspond to the signature output
+        $ curl http://127.0.0.1:5000/invocations -H 'Content-Type: application/json' -d '{
+            "columns": ["a", "b", "c"],
+            "data": [[1, 2, 3], [4, 5, 6]]
```

Comment:
Did you test this by installing mlflow from master branch locally? This is the 1.x request syntax (it will throw an exception now)
``` shell
{"error_code": "BAD_REQUEST", "message": "The input must be a JSON dictionary with exactly one of the input fields 
{'dataframe_split', 'instances', 'inputs', 'dataframe_records'}. Received dictionary with input fields: ['data']."}
```

Here are the new syntax options:

```shell
$ mlflow models serve -m runs:/my-run-id/model-path &

# records orientation input format for serializing a pandas DataFrame
$ curl http://127.0.0.1:5000/invocations -H 'Content-Type: application/json' -d '{
    "dataframe_records": [{"a":1, "b":2}, {"a":3, "b":4}, {"a":5, "b":6}]
}'

# split orientation input format for serializing a pandas DataFrame
$ curl http://127.0.0.1:5000/invocations -H 'Content-Type: application/json' -d '{
    "dataframe_split": {"columns": ["a", "b"],
                        "index": [0, 1, 2],
                        "data": [[1, 2], [3, 4], [5, 6]]}
}'

# inputs format for List submission of array, tensor, or DataFrame data
$ curl http://127.0.0.1:5000/invocations -H 'Content-Type: application/json' -d '{
    "inputs": [[1, 2], [3, 4], [5, 6]]
}'

# instances format for submission of Tensor data
curl http://127.0.0.1:5000/invocations -H 'Content-Type: application/json' -d '{
    "instances": [
        {"a": "t1", "b": [1, 2, 3]},
        {"a": "t2", "b": [4, 5, 6]},
        {"a": "t3", "b": [7, 8, 9]}
    ]
}'
```

Reply:
There is a dedicated test `test_model_predict_with_signature` which is passing after I merged the latest from master